application/common/components/notify/EmailServiceNotifier.php

```diff
@@ -0,0 +1,104 @@
+<?php
+namespace Sil\Idp\IdSync\common\components\notify;
+
+use InvalidArgumentException;
+use Sil\EmailService\Client\EmailServiceClient;
+use Sil\Idp\IdSync\common\interfaces\NotifierInterface;
+use Sil\Idp\IdSync\common\models\User;
+use yii\base\Component;
+
+/**
+ * NOTE: If you add public methods to this class, first add them to the
+ *       NotifierInterface.
+ */
+class EmailServiceNotifier extends Component implements NotifierInterface
+{
+    /**
+     * The array of configuration values for the email service client.
+     * @var array
+     */
+    public $emailServiceConfig;
+    
+    /**
+     * What address to send the email to.
+     * @var string 
+     */
+    public $emailTo;
+    
+    /**
+     * The name of the organization.
+     * @var string 
+     */
+    public $organizationName;
+    
+    public function init()
+    {
+        $this->assertConfigIsValid();
+        parent::init();
+    }
+    
+    protected function assertConfigIsValid()
+    {
+        $requiredParams = [
+            'accessToken',
+            'assertValidIp',
+            'baseUrl',
+            'validIpRanges',
+        ];
+        
+        foreach ($requiredParams as $param) {
+            if ( ! isset($this->emailServiceConfig[$param])) {
+                throw new InvalidArgumentException(
+                    'Missing ' . $param . ' value (for EmailServiceNotifier).',
+                    1502820156
+                );
+            }
+        }
+    }
+    
+    /**
+     * @return EmailServiceClient
+     */
+    protected function getEmailServiceClient()
+    {
+        $config = $this->emailServiceConfig;
+        return new EmailServiceClient(
+            $config['baseUrl'],
+            $config['accessToken'],
+            [
+                EmailServiceClient::ASSERT_VALID_IP_CONFIG => $config['assertValidIp'],
+                EmailServiceClient::TRUSTED_IPS_CONFIG => $config['validIpRanges'],
+            ]
+        );
+    }
+    
+    /**
+     * {@inheritdoc}
+     */
+    public function sendMissingEmailNotice(array $users)
+    {
```

Comment:
Since they're repeated, you could do this before and then put it in the render call ...
$templateVars = [
          'idStoreName' => $this->idStoreName,
          'organizationName' => $this->organizationName,
          'users' => $users,
]